src/jquery.big-slide.js

```diff
@@ -240,9 +288,22 @@
       toggleOpen: function() {
         settings.beforeOpen();
         controller.changeState();
-        this.$menu.css(settings.side, '0');
-        this.$push.css(settings.side, this.width);
-        this.$shrink.css('width', '100%').css('width', '-=' + this.$menu.width());
+        var screenWidth = $(window).width();
+        if (settings.semiOpenStatus && screenWidth > settings.semiOpenScreenWidth) {
+          this.$hiddenThin.show();
+          this.$menu.animate({ width: this.width}, {duration: Math.abs(settings.speed - 100), easing: 'linear'});
+          this.$push.css(settings.side, this.width);
+          this.$shrink.css({
+            'width': 'calc(100% - ' + this.width + ')'
+          });
+          this.$menu.removeClass('semiOpen');
+        } else {
+          this.$menu.css(settings.side, '0');
+          this.$push.css(settings.side, this.width);
+          this.$shrink.css({
+            'width': 'calc(100% - ' + this.width + ')'
+          });
```

Comment:
Could we move this as well as the code to style the closed menu to their own methods? Something like `applyOpenStyles` and `applyCloseStyles`:

For example:

```javascript

applyOpenStyles: function() {
 if (settings.semiOpenStatus && screenWidth > settings.semiOpenScreenWidth) {
    // ...
  } else {
    // ...
  }
}

toggleOpen: function() {
  settings.beforeOpen();	    
  controller.changeState();
  view.applyOpenStyles();
  menuLink.addClass(settings.activeBtn);
  settings.afterOpen();
  // save the state
  if (settings.saveState) {
    localStorage.setItem('bigSlide-savedState', 'open');
  }
}
```